support-frontend/assets/components/onboarding/heading.tsx

```diff
@@ -102,71 +100,131 @@ const thankYouSubtext = css`
 	}
 
 	${from.desktop} {
-		margin-bottom: ${space[4]}px;
+		margin-bottom: ${space[8]}px;
 	}
 `;
 
-interface OnboardingHeadingProps extends CSSOverridable {
+interface OnboardingHeadingProps extends OnboardingProps {
 	onboardingStep: OnboardingSteps;
 }
 
-function OnboardingHeading({ onboardingStep }: OnboardingHeadingProps) {
-	// --------------------------- //
-	// TODO: Dynamically generated based on Products and User profile
-	const ONBOARDING_STEP_CONTENT_MAP: Record<
-		OnboardingSteps,
-		{ headingHero: string; heading: string; subtext: string }
-	> = {
-		[OnboardingSteps.Summary]: {
-			headingHero: 'aspect ratio 21:9',
-			heading: 'Thank you for subscribing to All-access digital',
-			subtext:
-				"You've just joined over 1.3m others who support independent journalism.",
-		},
-		[OnboardingSteps.GuardianApp]: {
-			headingHero: 'Placeholder',
-			heading: 'Placeholder',
-			subtext: 'Placeholder',
-		},
-		[OnboardingSteps.FeastApp]: {
-			headingHero: 'Placeholder',
-			heading: 'Placeholder',
-			subtext: 'Placeholder',
-		},
-		[OnboardingSteps.ThankYou]: {
-			headingHero: 'Placeholder',
-			heading: 'Placeholder',
-			subtext: 'Placeholder',
-		},
-	};
-	// --------------------------- //
+function OnboardingHeading({
+	onboardingStep,
+	landingPageSettings,
+	productKey,
+}: OnboardingHeadingProps) {
+	const order = getThankYouOrder();
+	const productSettings =
+		productKey && landingPageSettings.products[productKey];
+
+	const { windowWidthIsGreaterThan, windowWidthIsLessThan } = useWindowWidth();
+
+	const ONBOARDING_STEP_CONTENT_MAP = useMemo<
+		Record<
+			OnboardingSteps,
+			{
+				heading?: string;
+				subtext?: string;
+				gridId?: string;
+				altText?: string;
+				aspectRatio: SerializedStyles;
+			}
+		>
+	>(() => {
+		// Non breaking hyphen
+		const allAccessTitle =
+			productSettings?.title === 'All-access digital'
+				? `All\u2011access digital`
+				: productSettings?.title;
```

Comment:
```suggestion
		const nonBreakingHyphen = '\u2011'
		const allAccessTitle =
			productSettings?.title === 'All-access digital'
				? `All${nonBreakingHyphen}access digital`
				: productSettings?.title;
```
or even more generally
```suggestion
		const nonBreakingHyphen = '\u2011'
		const allAccessTitle = productSettings?.title?.replaceAll('-', nonBreakingHyphen);
```